fvm/transactionStorageLimiter.go

```diff
@@ -14,29 +14,73 @@ import (
 
 type TransactionStorageLimiter struct{}
 
-func (_ TransactionStorageLimiter) CheckStorageLimits(
+// CheckStorageLimits checks each account that had its storage written to during a transaction, that its storage used
+// is less than its storage capacity.
+// Storage used is an FVM register and is easily accessible.
+// Storage capacity is calculated by the FlowStorageFees contract from the account's flow balance.
+//
+// The payers balance is considered to be maxTxFees lower that its actual balance, due to the fact that
+// the fee deduction step happens after the storage limit check.
```

Comment:
```suggestion
// CheckStorageLimits reads the storage usage of each account and the payer, 
// and reads their storage capacity from the given environment, and check if the storage
// usage of each account exceeds the storage capacity. 
// In additional, it also check if the given payer has enough balance to pay for the given maxTxFees
// It returns StorageCapacityExceededError if storage usage of a certain account exceeds its storage capacity
// It returns ErrCodeInsufficientPayerBalance if the payer has insufficient balance to cover maxTxFees.
// Storage used is an FVM register and is easily accessible without cadence.
// Storage capacity is calculated by the FlowStorageFees contract from the account's flow balance.
//
// The payers balance is considered to be maxTxFees lower that its actual balance, due to the fact that
// the fee deduction step happens after the storage limit check.
```

Reply:
```
// In additional, it also check if the given payer has enough balance to pay for the given maxTxFees
...
// It returns ErrCodeInsufficientPayerBalance if the payer has insufficient balance to cover maxTxFees.
```
I different function does this part: `CheckPayerBalanceAndReturnMaxFees`